util/file.py

```diff
@@ -94,6 +94,27 @@ def mkstempfname(suffix='', prefix='tmp', directory=None, text=False):
     return fn
 
 
+@contextlib.contextmanager
+def tempfname(*args, **kwargs):
+    '''Create a tempfile name on context entry, delete the file (if it exists) on context exit.'''
+    fn = mkstempfname(*args, **kwargs)
+    try:
+        yield fn
+    finally:
+        if os.path.isfile(fn): os.unlink(fn)
+
+
+@contextlib.contextmanager
+def tempfnames(suffixes, *args, **kwargs):
+    '''Create a setof tempfile names on context entry, delete the files (if they exist) on context exit.'''
```

Comment:
"we're trying to follow pep8" -- ok.   I prefer using fewer lines of code so more of it fits on the screen at once.  I also find that more spaces in a line make it easier to read.  But ok, standards are standards.